### DISCUSSION TITLE
Relationship field and array limit - Errors: 429 (Too Many Requests) and 431 (Request Header Fields Too Large)

### DISCUSSION THREAD

**Discussion by vstarush** [payloadcms/payload#1977] [Q&A]
Hello,

Is there a limit to how many item can be in a Relationship and Array fields?

For 260+ items in a relationship field I am getting the **431 (Request Header Fields Too Large)** error.

And also, if I create an array component with a relationship field as the only field, I am getting the **429 (Too Many Requests)** error.

Is there a way to have 300+ items within a relationship field or in the array component with a single relationship field?

Thank you,
Vitaly

**Comment by DanRibbens** [comment]
Hey @vstarush,

I haven't tested with as many as you're saying so I haven't seen this before. Payload doesn't impose any limits, so what you're seeing must be a browser / http limitation.

What is the endpoint is hitting this header size limit error? Can you share what the request looks like from your browser network tab and we can troubleshoot and possibly optimize how the request is made.

Also sharing your payload config or a reproduction repo can also help me understand what you're trying to model and we can come up with a workaround.

Thanks!

**Reply by andershermansen** [reply]
@DanRibbens I got hit by this issue today.

The issue is that when you want to add a new item to a relationship field, it will do a HTTP call which will fetch the available items. The where in this call will be dynamically created to not include the already selected items.

The query is here:
https://github.com/payloadcms/payload/blob/fd02bee0fee3ab1dde92dffbd08329cf880bb051/packages/ui/src/fields/Relationship/index.tsx#L165C1-L203C15

If you have many items the where will be very long. If you have enough items the generated query will exceed what is allowed by browser/http.

![Skjermbilde 2024-05-24 kl  09 50 28](https://github.com/payloadcms/payload/assets/2082176/bf686442-0591-49be-be3b-120a4717ca06)

![Skjermbilde 2024-05-24 kl  09 50 52](https://github.com/payloadcms/payload/assets/2082176/1ba0c602-a23e-4d6d-a4f0-bbadec66700a)

**Reply by andershermansen** [reply]
I have implemented a fix here https://github.com/payloadcms/payload/pull/6487

Hope you like the approach of using X-HTTP-Method-Override

**Reply by andershermansen** [reply]
This fix is now part of [v3.0.0-beta.47](https://github.com/payloadcms/payload/releases/tag/v3.0.0-beta.47)

**Comment by BarrySaikSoundry** [comment]
I have found that this feature breaks hosting on AWS  cloudfront. Cloudfront removes the X-HTTP-Method-Override header as a security measure and all the relationship fields do not load correctly in the UI because the POST request to /api/{collection} now fails. Huge showstopper for me, unfortunately.

**Reply by andershermansen** [reply]
It seems you can whitelist headers? https://aws.amazon.com/blogs/aws/enhanced-cloudfront-customization/

**Reply by BarrySaikSoundry** [reply]
Yea - I tried to get that to work. I wish that worked with Amplify. Amplify hides the cloudfront distribution it creates to provision your app from what I can tell.
